src/ng/compile.js

```diff
@@ -3101,6 +3152,33 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
         }
       });
 
+      function recordChanges(key, currentValue, previousValue) {
+        if (isFunction(destination.$onChanges) && currentValue !== previousValue) {
+          // If we have not already scheduled the top level onChangesQueue handler then do so now
+          if (!onChangesQueue) {
+            scope.$$postDigest(flushOnChangesQueue);
```

Comment:
No need anymore FWIW.  The addition of the other TTL mechanism mitigated most of my concern, though it'd be sweet if the existing TTL could have been passed along rather then implementing another one so that the max TTL can be defined in one place.